### ISSUE TITLE
Caffe doesn't support NVidia Tesla V100 GPUs

### ISSUE THREAD

**Issue by ghostplant** [BVLC/caffe#6531]
### Issue summary

When running Caffe binary applications on V100 GPUs, the application returns CUDA error = CUDA_ERROR_NO_BINARY_FOR_GPU. However, legacy models like P100 GPUs and K80c GPUs could execute Caffe correct. This issue even happens when building Caffe from source code directory on V100 nodes.

### Steps to reproduce

On nodes with V100 GPUs, build from source code, run typical applications and it fails for CUDA_ERROR_NO_BINARY_FOR_GPU.

### System configuration

* Operating system: 
Ubuntu 16.04 LTS
* Compiler: 
nvcc
* CUDA version (if applicable): 
CUDA 9.2
* CUDNN version (if applicable): 
CUDNN 7.2.1
* BLAS: 
* Python version (if using pycaffe): 
* MATLAB version (if using matcaffe): 

### Issue checklist

- [Y] read the guidelines and removed the first paragraph
- [Y] written a short summary and detailed steps to reproduce
- [Y] explained how solutions to related problems failed (tick if found none)
- [Y] filled system configuration
- [Y] attached relevant logs/config files (tick if not applicable)

**Comment by yyyreal** [comment]
**USE CUDA 9 AND CUDNN 7 PLEASE**
![image](https://user-images.githubusercontent.com/13974360/45485174-3ad15500-b789-11e8-9b30-54c8c51f633a.png)
Caffe works well on my machine.

**Comment by ghostplant** [comment]
Yes, but it is strange that any other platforms like Tensorflow, Pytorch, MXNet, CNTK don't have this limitation and they can build and run using CUDA 8.0 + V100.

I think there are some incorrect nvcc compiling configurations that make the build and execution fails for Caffe.

**Comment by IvyGongoogle** [comment]
@yyyreal  I meet the same error on my `tesla v100 gpu`, and then I use `cuda9.1 + cudnn7`, it still not works.   when I run `./examples/mnist/train_lenet.sh`, it stucks in:

```
I1116 10:48:17.978020 95005 layer_factory.hpp:77] Creating layer conv1
I1116 10:48:17.978057 95005 net.cpp:86] Creating Layer conv1
I1116 10:48:17.978072 95005 net.cpp:408] conv1 <- data
I1116 10:48:17.978096 95005 net.cpp:382] conv1 -> conv1
```

can anyone give some advises?

**Comment by IvyGongoogle** [comment]
finally, it starts to train, but it is so slow.....

**Comment by yyyreal** [comment]
> finally, it starts to train, but it is so slow.....

I suggest you to check whether incompatible libraries you used when you compiled caffe, since one of my colleague met the same issue once.

**Comment by Luckick** [comment]
> finally, it starts to train, but it is so slow.....

Hi, 

Could you explain how did you make it work? Is the slow due to use CPU mode?
Thanks!

**Comment by huyutao3550346** [comment]
Has the problem been sloved？
I run the caffe in the  Tesla K80 and the effect is well,It takes about 40ms every times.
But when I run the code in the Tesla P4,the time increase to 500ms.
Both of code are the same(use GPU mode).
I modifiy the CmakeList.txt,append 60 61 in the CMAKE_CUDA_FLAGS_RELEASE
set( CMAKE_CUDA_FLAGS_RELEASE "${CMAKE_CUDA_FLAGS_RELEASE}  -gencode arch=compute_30,code=sm_30 -gencode arch=compute_35,code=sm_35 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_52,code=sm_52 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_61,code=sm_61 -gencode arch=compute_61,code=compute_61")
however it isn't work.
